lib/mix/lib/mix/dep.ex

```diff
@@ -167,10 +167,10 @@ defmodule Mix.Dep do
     do: "lock mismatch: the dependency is out of date (run \"mix deps.get\" to fetch locked version)"
```

Comment:
"out of date" does not refer to point in time of the new dependency, it refers to the current lock file. Imagine I am using ecto 0.2.0 and then I go to 0.1.0, my dependency is out of date with the current version in the lock, it is nothing about which ecto version.